### PR TITLE
Burst handler improvements

### PR DESCRIPTION
**Pull request by cahz**
* Fix off by one bug in last signal in RMW/WriteBurstHandler
* Increase maximum size of write transfers and error when exceeding

### CODE REVIEW

File: src/RMWBurstHandler.bsv [hunk 1]
```diff
@@ -165,7 +165,7 @@ module mkRMWBurstHandler(RMWBurstHandler);
 
         if(last_in_packet || beat_cntr_t + 1 == 0) begin
             Bool last_cntr_set = False;
-            if(last_cntr == fromInteger(valueOf(MaximumAXIBeats))) begin
+            if(last_cntr == fromInteger(valueOf(MaximumAXIBeats)) - 1) begin
```
Comment (jahofmann) [inline src/RMWBurstHandler.bsv:168]: Thanks for this. Looks good.

File: src/WriteBurstHandler.bsv [hunk 2]
```diff
@@ -176,7 +176,7 @@ module mkWriteBurstHandler(WriteBurstHandler);
 
         if(flit_cntr_t == 0 || beat_cntr_t + 1 == 0) begin
             Bool last_cntr_set = False;
-            if(last_cntr == fromInteger(valueOf(MaximumAXIBeats))) begin
+            if(last_cntr == fromInteger(valueOf(MaximumAXIBeats)) - 1) begin
```
Comment (jahofmann) [inline src/WriteBurstHandler.bsv:179]: Same as above.

File: src/WriteBurstHandler.bsv [hunk 3]
```diff
@@ -246,9 +246,14 @@ module mkWriteBurstHandler(WriteBurstHandler);
             outstanding_requests.enq(tuple2(escapeAddrForAXI(addr), cExtend(beats)));
         end
 
-        // Assuming 8192 Bytes as the Maximum -> 2 AXI Requests
+        // Assuming 12288 Bytes as the Maximum -> 3 AXI Requests
         let requests = 1;
-        if(beats > fromInteger(valueOf(MaximumAXIBeats))) begin
+        if(beats > fromInteger(3*valueOf(MaximumAXIBeats))) begin
```
Comment (jahofmann) [inline src/WriteBurstHandler.bsv]: For the reader: This is needed as cahz has some additional data to transfer. The OmniXtend endpoint supports up to 8 kB transfers right now to align with Ethernet jumbo packets.

I wonder if this adds hardware in cases where we don't need it. We should consider a generic function instead. For instance, we could create a LUT using a for-loop to support all sizes.

**Review by jahofmann — CHANGES_REQUESTED** [review]
I'd prefer if this function is more generic. The way it is done right now it might add unnecessary overhead in a pure OX/TL application.

File: src/WriteBurstHandler.bsv (hunk 3 above)
Comment (cahz) [reply src/WriteBurstHandler.bsv]: I will have a look at it

Comment (cahz) [reply src/WriteBurstHandler.bsv]: I have pushed an update. A first look at synthesis results don't show an increased overhead.

**Review by jahofmann — APPROVED** [review]
Thanks for the generalized version.

**Pull request merged**